### ISSUE TITLE
Extend it to support TFHE encryption scheme

### ISSUE THREAD

**Issue by enricobottazzi** [privacy-scaling-explorations/greco#28]
TFHE still works on RLWE so it shouldn't be too hard to adapt it -> https://www.zama.ai/post/tfhe-deep-dive-part-1

**Comment by adust09** [comment]
The back-end FHE primitives will be important👍

The purpose of this issue is to support multiple primitives, but what is the position of bfv-py ?

The readme says it is for educational purposes, but can we recognise this repository as an official BFV implementation in Greco?

**Comment by enricobottazzi** [comment]
Hi @adust09. `bfv-py` is a very inefficient implementation of bfv algorithm that I made and that Greco happens to use for testing the whole flow. In particular, it is used to generate the inputs for the circuit. Any bfv backend should also be compatible with the existing circuit as long as you build some script (similar to [this](https://github.com/privacy-scaling-explorations/greco/blob/main/scripts/circuit_sk.py)) to properly format the input and the parameters. 

Not sure if this answers your question. What was your concern with bfv-py?

**Comment by adust09** [comment]
MAKE SENSE!
I personally found the implementation of bfv-py incomplete and wondered if it was a suitable backend for greco.For example, encording (currently only valid for sample_polynomial) and EvalMul are not implemented.So I wondered if there was a plan to replace them with other BFV implementations before TFHE.

**Comment by adust09** [comment]
btw, I've always wondered why it decrypts immediately after encrypting with circuit_sk without any operation (EvalAdd in bfv-py).I was satisfied with the explanation that this is also not directly related to the overall Greco flow test, so it is not executed.

**Comment by enricobottazzi** [comment]
Exactly, you shouldn't see bfv-py as a backend, as there's really no backend needed here. 

> So I wondered if there was a plan to replace them with other BFV implementations before TFHE.

There's no particular priority here, it all depends if there's demand or willingness to contribute

**Comment by Vishalkulkarni45** [comment]
I would love to build the TFHE support for greco ,can i take this issue?
